Tools/cases_generator/stacking.py

```diff
@@ -261,15 +261,19 @@ def adjust_higher(self, eff: StackEffect) -> None:
         self.final_offset.higher(eff)
 
     def adjust(self, offset: StackOffset) -> None:
-        for down in offset.deep:
+        deep = list(offset.deep)
+        high = list(offset.high)
+        for down in deep:
             self.adjust_deeper(down)
-        for up in offset.high:
+        for up in high:
             self.adjust_higher(up)
```

Comment:
Just curious, why? I assume it has something to do with the comment you removed below (`# Use clone() since adjust_inverse() mutates final_offset`), but I'm not sure I understand.

Reply:
Indeed. That `.clone()` call was needed because `adjust_higher()` and `adjust_deeper()` change `self.final_offset` (in place), and the argument here may be `self.final_offset`. (And ditto for `min_offset`.) So previously my "fix" for this was to pass in a clone of the argument. The new fix avoids the need to clone in the caller but effectively clones the `deep` and `high` lists of the argument.